besca/tl/rc/_reclustering.py

```diff
@@ -90,21 +93,27 @@ def recluster(adata,
 
     cluster_subset.raw = cluster_subset
 
-    #identify highly variable genes
-    filter_result = sc_highly_variable_genes(cluster_subset, min_mean = min_mean,
-                                             max_mean = max_mean, min_disp = min_disp,  inplace=False, batch_key=batch_key)
+    #identify highly variable genes    
+    sc_highly_variable_genes(cluster_subset, min_mean=min_mean, max_mean=max_mean, min_disp=min_disp, 
+                                             inplace=True, batch_key=batch_key)
+    
+    if (batch_key!=None):
+        hvglist=cluster_subset.var['highly_variable'].copy()
+        hvglist.loc[cluster_subset.var['highly_variable_nbatches']>=len(set(cluster_subset.obs[batch_key]))/n_shared,]=True
+        cluster_subset.var['highly_variable']=hvglist.copy()
+        
     if show_plot_filter:
-        plot_filter(filter_result)
-    print('In total', str(sum(filter_result.highly_variable)), 'highly variable genes selected within cluster')
+        pl_highly_variable_genes(cluster_subset, show=True)
+    print('In total', str(sum(cluster_subset.var.highly_variable)), 'highly variable genes selected within cluster')
 
     #apply filter
-    cluster_subset = _subset_adata(cluster_subset, filter_result.highly_variable, axis = 1, raw = False)
+    cluster_subset = _subset_adata(cluster_subset, cluster_subset.var.highly_variable, axis = 1, raw = False)
 
     #perform further processing
-    log1p(cluster_subset)
+    # log1p(cluster_subset) # data already logged 
```

Comment:
We could remove the line. It would be clearer ?

Reply:
I would keep it, just if in case we missed something here